core/new-gui/src/app/workspace/service/joint-ui/joint-ui.service.spec.ts

```diff
@@ -0,0 +1,142 @@
+import * as joint from 'jointjs';
+
+import { JointUIService } from './joint-ui.service';
+import { OperatorMetadataService } from '../operator-metadata/operator-metadata.service';
+import { StubOperatorMetadataService } from '../operator-metadata/stub-operator-metadata.service';
+import { MOCK_OPERATOR_METADATA } from '../operator-metadata/mock-operator-metadata.data';
+
+describe('JointUIService', () => {
+  let service: JointUIService;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [
+        JointUIService,
+        { provide: OperatorMetadataService, useClass: StubOperatorMetadataService },
+      ],
+    });
+  });
+
+  it('should be created', inject([JointUIService], (ser: JointUIService) => {
+    service = ser;
+    expect(service).toBeTruthy();
+  }));
+
+  /**
+   * Check if the getJointjsOperatorElement() can successfully creates a JointJS Element
+   */
+  it('getJointjsOperatorElement() should create an operatorElement', () => {
+    const result = service.getJointjsOperatorElement('ScanSource', 'operator1', 100, 100);
+    expect(result).toBeTruthy();
+  });
+
+  /**
+   * Check if the error in getJointjsOperatorElement() is correctly thrown
+   */
+  it('getJointjsOperatorElement() should throw an error', () => {
+    const nonExistingOperator = 'NotExistOperator';
+    expect(
+      function() {
+        service.getJointjsOperatorElement(nonExistingOperator, 'operatorNaN', 100, 100);
+      }
+    )
+    .toThrow(new Error('JointUIService.getJointUI: ' +
+    'cannot find operatorType: ' + nonExistingOperator));
+  });
+
+
+  /**
+   * Check if the number of inPorts and outPorts created by getJointjsOperatorElement()
+   * matches the port number specified by the operator metadata
+   */
+  it('getJointjsOperatorElement() should create correct number of inPorts and outPorts', () => {
+    const element1 = service.getJointjsOperatorElement('ScanSource', 'operator1', 100, 100);
+    const element2 = service.getJointjsOperatorElement('NlpSentiment', 'operator1', 100, 100);
+    const element3 = service.getJointjsOperatorElement('ViewResults', 'operator1', 100, 100);
+
+    const inPortCount1 = element1.getPorts().filter(port => port.group === 'in').length;
+    const outPortCount1 = element1.getPorts().filter(port => port.group === 'out').length;
+    const inPortCount2 = element2.getPorts().filter(port => port.group === 'in').length;
+    const outPortCount2 = element2.getPorts().filter(port => port.group === 'out').length;
+    const inPortCount3 = element3.getPorts().filter(port => port.group === 'in').length;
+    const outPortCount3 = element3.getPorts().filter(port => port.group === 'out').length;
+
+    expect(inPortCount1).toEqual(0);
+    expect(outPortCount1).toEqual(1);
+    expect(inPortCount2).toEqual(1);
+    expect(outPortCount2).toEqual(1);
+    expect(inPortCount3).toEqual(1);
+    expect(outPortCount3).toEqual(0);
+
+  });
+
```

Comment:
Should there also be a test to test that linking of two elements works?

Reply:
Since JointJS are managing all the links, there's no point of testing the correctness of JointJS. We trust the library is implemented correctly.